crates/augurs-forecaster/src/power_transforms.rs

```diff
@@ -0,0 +1,70 @@
+use crate::transforms::box_cox;
+use argmin::core::*;
+use argmin::solver::brent::BrentOpt;
+
+fn box_cox_log_likelihood(data: &[f64], lambda: f64) -> f64 {
+    let n = data.len() as f64;
+    let transformed_data: Vec<f64> = data.iter().map(|&x| box_cox(x, lambda)).collect();
+    let mean_transformed: f64 = transformed_data.iter().copied().sum::<f64>() / n;
+    let variance: f64 = transformed_data
+        .iter()
+        .map(|&x| (x - mean_transformed).powi(2))
+        .sum::<f64>()
+        / n;
+
+    // Avoid log(0) by ensuring variance is positive
+    let log_likelihood =
+        -0.5 * n * variance.ln() + (lambda - 1.0) * data.iter().map(|&x| x.ln()).sum::<f64>();
+    log_likelihood
+}
+
+#[derive(Clone)]
+struct BoxCoxProblem {
+    data: Vec<f64>,
```

Comment:
We might be able to get away with a `&[f64]` here to avoid converting the slice to a vec in `optimizer_lambda`.

Reply:
Ah, something like...

```rust
#[derive(Clone)]
struct BoxCoxProblem<'a> {
    data: &'a [f64],
}

impl CostFunction for BoxCoxProblem<'_> {

...
```

(we can elide the explicit lifetime name from the `CostFunction` impl because we don't actually refer to it in the impl, otherwise we'd use `impl<'a> CostFunction for BoxCoxProblem<'a>`)